lib/drivers/carelinkDriver.js

```diff
@@ -201,6 +203,52 @@ module.exports = function(simulatorMaker, api){
         var uploads = removeOverlaps(payload);
         progress(20);
 
+        function getUnits() {
+          var units = null;
```

Comment:
Assuming that the colNames stuff has been called before you call getUnits (and it looks like it has) couldn't you just extract the units from colNames.CalBGForPH without bothering to do this search?